persistence/mongo/src/main/scala/za/co/absa/spline/persistence/mongo/dao/LineageDAOv4.scala

```diff
@@ -100,6 +100,15 @@ class LineageDAOv4(override val connection: MongoConnection) extends BaselineLin
             addAll(opTransformations.asJava)
           })
         }
+        else if (isWriteOperation(op)) {
+          if (!op.containsField("writeMetrics")) {
```

Comment:
In operations_v4 some writes have metrics and some don't?

Reply:
Correct. Metrics were added to the model without changing the model version. The assumption was that since metrics are optional absence of a field would be treated as an empty field (empty map). But in fact Salat doesn't treat is like that. So the purpose of this line of code is to force that default.